src/components/icon/icon.ts

```diff
@@ -18,18 +21,33 @@ const iconCache = new Map<string, Promise<SVGResult>>();
  * @status stable
  * @since 2.0
  *
- * @event sl-load - Emitted when the icon has loaded.
- * @event sl-error - Emitted when the icon fails to load due to an error.
+ * @event sl-load - Emitted when the icon has loaded. When using `spriteSheet: true` this will not emit.
+ * @event sl-error - Emitted when the icon fails to load due to an error. When using `spriteSheet: true` this will not emit.
  *
  * @csspart svg - The internal SVG element.
+ * @csspart use - The <use> element generated when using `spriteSheet: true`
  */
 @customElement('sl-icon')
 export default class SlIcon extends ShoelaceElement {
   static styles: CSSResultGroup = styles;
 
+  private initialRender = false;
+
   /** Given a URL, this function returns the resulting SVG element or an appropriate error symbol. */
-  private static async resolveIcon(url: string): Promise<SVGResult> {
+  private async resolveIcon(url: string, library?: IconLibrary): Promise<SVGResult> {
     let fileData: Response;
+
+    if (library?.spriteSheet) {
+      return html`<svg
+        part="svg"
+        ${'' /* In testing, these dont seem to actually do anything. */}
+        @error=${() => this.emit('sl-error')}
+        @load=${() => this.emit('sl-load')}
```

Comment:
We're documenting that these aren't emitted with `spriteSheet`. If that's true, we can probably drop these lines.

Reply:
Sounds good to me.